### PR TITLE
fixed ignored error from getRequestState()

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
re: e-support CZN-403791
AbstractServlet.getRequestState is throwing
a null pointer exception and it is being ignored.
Modify to properly report failures in getRequestState.
Does not fix cause of the failure, but at least it gets reported now.

### CODE REVIEW

File: opendap/src/main/java/opendap/servlet/AbstractServlet.java [hunk 1]
```diff
@@ -413,19 +413,22 @@ public void IOExceptionHandler(IOException e,  ReqState rs) {
    * @param e        The exception that caused the problem.
    * @param rs The <code>ReqState</code> for the client.
    */
-  public void anyExceptionHandler(Throwable e, ReqState rs) {
-    HttpServletResponse response = rs.getResponse();
+  public void anyExceptionHandler(Throwable e, ReqState rs)
+  {
+    log.error("DODServlet ERROR (anyExceptionHandler): " + e);
+    e.printStackTrace();
+    printThrowable(e);
```
Comment (cwardgar) [inline opendap/src/main/java/opendap/servlet/AbstractServlet.java]: I dislike adding new code that prints to STDOUT or STDERR. Why not just use [this method](https://www.slf4j.org/api/org/slf4j/Logger.html#error(java.lang.String,%20java.lang.Throwable)), which will print the entire exception message and stack trace to the log? Better still would be if you removed *all* printing to STDERR from the whole class.

Comment (cwardgar) [inline opendap/src/main/java/opendap/servlet/AbstractServlet.java]: Now that I think about it, I'm not completely sure that that method will print the exception's stack trace to the log. Maybe 95% sure. Wouldn't hurt to test.

**Comment by DennisHeimbigner** [comment]
Agree about changing to use log; but this code is same as before, just moved location.

**Comment by cwardgar** [comment]
Just moved location? I don't understand. Is there some harm in making those changes? Seems pretty straightforward.

**Comment by cwardgar** [comment]
Oh, you're saying that the code isn't exactly "new". Okay, but those changes would still be good.

**Comment by cwardgar** [comment]
Can you rebase this?

**Comment by DennisHeimbigner** [comment]
ok, up to date

**Comment by cwardgar** [comment]
We've got conflicts. Possibly because you merged in master instead of rebasing on top of it? I dunno.

**Comment by DennisHeimbigner** [comment]
I did rebase it. Why does the pull request say there are no conflicts?

**Comment by cwardgar** [comment]
Not sure, but I see a merge commit (https://github.com/Unidata/thredds/pull/804/commits/7eaaaaac3f4cab211e42ff51642340edbfdd7dc2) and a PR in conflict: 
screenshot from 2017-04-15 19-08-00
Do you see something diffrent?

**Comment by DennisHeimbigner** [comment]
Yes.ee atached
[pr804.pdf](https://github.com/Unidata/thredds/files/924021/pr804.pdf)

**Comment by cwardgar** [comment]
Well, that's weird as hell. I wonder what @lesserwhirls sees.

Still, even on your end, I see a merge commit, as well as the inclusion of commits from me and Sean in the PR. That tells me that something went wrong with your rebase. Try the rebase again, maybe? Possibly in a new PR.

**Comment by DennisHeimbigner** [comment]
ok I will see what I can do.
=Dennis

**Comment by DennisHeimbigner** [comment]
Ok, I started over and created pr 806 from most recent version
of master + changes from 804.
Hope that fixes things.
=Dennis

**Comment by cwardgar** [comment]
Superseded by #806.

**Pull request closed without merging**